manifests/extpack.pp

```diff
@@ -0,0 +1,67 @@
+# == Class: virtualbox::extpack
+#
+# This class (un)installs the VirtualBox extension pack.
+#
+class virtualbox::extpack (
+  $ensure  = $virtualbox::params::extpack_ensure,
+  $version = $virtualbox::params::extpack_version
+) {
+  validate_string($ensure)
+  validate_string($version)
+
+  if $version !~ /[^r]+r[^r]+/ {
+    fail("'${version}' is not a valid version string!")
+  }
+
+  case $ensure {
+    'present': {
+      if $virtualbox::package_ensure == 'absent' {
+        fail('Cannot install extension pack without installing VirtualBox')
+      }
+
+      # install/upgrade if we do not have the requested version
+      if $virtualbox_extpack_version != $version {
+        $ver = regsubst($version, '^([^r]+)r([^r]+)', '\1')
+        $build = regsubst($version, '^([^r]+)r([^r]+)', '\2')
+        $filename = "Oracle_VM_VirtualBox_Extension_Pack-${ver}-${build}.vbox-extpack"
+        $url = "http://download.virtualbox.org/virtualbox/${ver}/${filename}"
+        $dl_dir = '/tmp'
+
+        wget::fetch { 'download vbox extpack':
+          source      => $url,
+          destination => "${dl_dir}/${filename}",
+        }
+
+        exec { 'install vbox extpack':
+          command => "VBoxManage extpack install --replace ${dl_dir}/${filename}",
+          path    => '/usr/local/bin:/usr/bin:/bin',
+          require => [
+            Class['virtualbox::install'],
+            Wget::Fetch['download vbox extpack'],
+          ],
+        }
+      }
+    }
+    'absent': {
+      if $virtualbox_extpack_version {
+        exec { 'uninstall vbox extpack':
+          command => 'VBoxManage extpack uninstall "Oracle VM VirtualBox Extension Pack"',
+          path    => '/usr/local/bin:/usr/bin:/bin',
+        }
+
+        # Make sure we process uninstall request after installing VirtualBox
+        # (if VirtualBox will be installed during the same Puppet run) and
+        # before removing VirtualBox (if VirtualBox will be uninstalled during
+        # the same Puppet run)
+        if $virtualbox::package_ensure == 'absent' {
+          Exec['uninstall vbox extpack'] -> Class['virtualbox::install']
+        } else {
+          Class['virtualbox::install'] -> Exec['uninstall vbox extpack']
+        }
+      }
+    }
+    'default': {
+      fail('Invalid value for attribute \'ensure\'')
```

Comment:
This can be written as: `fail("Invalid value for attribute 'ensure'")`

Double quotes are ok for interpolation or for cases where you need to use single quotes. I'm not sure if `puppet-lint` picks up on this. If not, we can tell it to ignore this line.